XamarinCommunityToolkit/Helpers/SafeAreaTypeConverter.shared.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using Xamarin.Forms;
+
+namespace XamarinCommunityToolkit.Helpers
+{
+    [Xamarin.Forms.Xaml.TypeConversion(typeof(SafeArea))]
+    public class SafeAreaTypeConverter : TypeConverter
+    {
+        public override object ConvertFromInvariantString(string value)
+        {
+            if (value != null)
+            {
+                value = value.Trim();
+
+                if (value.Contains(","))
+                {
+                    // XAML based definition
+                    var safeArea = value.Split(',');
+
+                    switch (safeArea.Length)
+                    {
+                        case 2:
+                            if (bool.TryParse(safeArea[0], out var h)
+                                && bool.TryParse(safeArea[1], out var v))
+                                return new SafeArea(h, v);
+                            break;
+                        case 4:
+                            if (bool.TryParse(safeArea[0], out var l)
+                                && bool.TryParse(safeArea[1], out var t)
+                                && bool.TryParse(safeArea[2], out var r)
+                                && bool.TryParse(safeArea[3], out var b))
+                                return new SafeArea(l, t, r, b);
+                            break;
+                    }
+                }
+                else
+                {
+                    // Single uniform SafeArea
+                    if (bool.TryParse(value, out var l))
+                        return new SafeArea(l);
+                }
+            }
+
+            throw new InvalidOperationException($"Cannot convert \"{value}\" into {typeof(SafeArea)}");
```

Comment:
I believe it can be more laconic :)
```suggestion
            try
            {
                var safeAreaItems = value?.Trim().Split(',').Select(bool.Parse);
                return safeAreaItems.Length switch
                {
                    1 => new SafeArea(l),
                    2 => new SafeArea(h, v),
                    4 => new SafeArea(l, t, r, b),
                    _ => throw new Exception()
                };
            }
            catch
            {
                throw new InvalidOperationException($"Cannot convert \"{value}\" into {typeof(SafeArea)}");
            }
```

Reply:
The problem I have is that sometimes shorter is not necessarily more readable 😂

Reply:
Also, in your suggestion, where are all the variables coming from like `l`, `h`, `v`, `t`, `r` and `b`? I would suppose the LINQ query select creates a dynamic list, but I'd have to grab them by index? Not sure if I like that more.

Reply:
> Also, in your suggestion, where are all the variables coming from like `l`, `h`, `v`, `t`, `r` and `b`? I would suppose the LINQ query select creates a dynamic list, but I'd have to grab them by index? Not sure if I like that more.

yes, I meant getting by the index of course )

Reply:
Then

```csharp
public override object ConvertFromInvariantString(string value)
{
    var items = ParseArray(value);
    return items.Length switch
    {
        1 => new SafeArea(items[0]),
        2 => new SafeArea(items[0], items[1]),
        4 => new SafeArea(items[0], items[1], items[2], items[3]),
        _ => throw new InvalidOperationException($"Cannot convert \"{value}\" into {typeof(SafeArea)}")
    };
}

bool[] ParseArray(string value)
{
    var source = value?.Trim().Split(',');
    var items = new bool[source?.Length ?? 0];
    for (var i = 0; i < items.Length; i++)
    {
        if (!bool.TryParse(source[i], out var item))
            return null;

        items[i] = item;
    }
    return items;
}
```

Reply:
I think I agree with Steven for this. Yes, the solution is technically accurate and will work fine, but I think what's in there now is more readable for the time being. Let's just start out simple and engineer whenever we have a need for it not because we might need to have to in the future.